### PR TITLE
[DROOLS-3367] Created DMNRuntimeFluent and related APIs

### PR DESCRIPTION
**Pull request by danielezonca**
Implement DMN fluent API plus small changes of existing API

It supports at the moment only a single run definition per time because it registers `DMNModel`, `DMNContext` etc without any concept like "DMN session id". I think it is not really needed because DMN is stateless so we can assume user are working only on a single model per time but it can be easily updated in the future.

@mariofusco @tarilabs @kkufova  

PRs list
https://github.com/kiegroup/droolsjbpm-knowledge/pull/350
https://github.com/kiegroup/drools/pull/2179

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/SetDMNInputCommand.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        this.name = name;
+        this.value = value;
+    }
+
+    @Override
+    public Void execute(Context context) {
+        RegistryContext registryContext = (RegistryContext) context;
+        DMNContext dmnContext = registryContext
+                .computeIfAbsent(DMNContext.class,
+                                 clazz -> DMNFactory.newContext());
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/SetDMNInputCommand.java]: here you may want to create a DMNContext actually from DMNRuntime instead

**Review by tarilabs — APPROVED** [review]

File: kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/SetDMNInputCommand.java (hunk 1 above)
Comment (danielezonca) [reply kie-dmn/kie-dmn-core/src/main/java/org/kie/dmn/core/fluent/SetDMNInputCommand.java]: Done :)

**Review by kkufova — APPROVED** [review]
Looks good to me, except for the `return null;` statement. :-)

**Comment by danielezonca** [comment]
jenkins execute compile downstream build

**Pull request merged**